src/core/pointcloud/qgspointcloudlayerexporter.h

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+                         qgspointcloudlayerexporter.h
+                         ---------------------
+    begin                : July 2022
+    copyright            : (C) 2022 by Stefanos Natsis
+    email                : uclaros at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOINTCLOUDLAYEREXPORTER_H
+#define QGSPOINTCLOUDLAYEREXPORTER_H
+
+#include "qgis_core.h"
+#include "qgsvectorlayer.h"
+#include "qgspointcloudlayer.h"
+
+/**
+ * \class QgsPointCloudLayerExporter
+ * \ingroup core
+ *
+ * \brief Handles exporting PointCloud layers to Temporary scratch layers, OGR supported files and PDAL supported files
+ *
+ * \since QGIS 3.28
+ */
+class CORE_EXPORT QgsPointCloudLayerExporter
+{
+  public:
+
+    /**
+     * Constructor for QgsPointCloudLayerExporter, associated with the specified \a layer.
```

Comment:
Can you add some description about the required lifetime of `layer`?

Ideally the exporter would copy everything it needs in the constructor so that the layer can be destroyed before the exporter is used, as that would make it easy to run in a task in a thread-safe way.